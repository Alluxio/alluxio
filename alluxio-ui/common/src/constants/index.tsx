export * from './types/INavigationData';
export * from './types/IFileBlockInfo';
export * from './types/IFileInfo';
