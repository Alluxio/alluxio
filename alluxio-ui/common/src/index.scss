/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

@import './constants/scss/themeVariables';
@import '~bootstrap/scss/bootstrap';
@import '~bootstrap/scss/bootstrap-grid';
@import '~bootstrap/scss/bootstrap-reboot';
@import './constants/scss/themeCustomization';
@import '~source-sans-pro/source-sans-pro.css';

body {
  margin: 0;
  padding: 0;
  font-family: 'Source Sans Pro', sans-serif;
  font-size: 1em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-shadow: rgba(0,0,0,.01) 0 0 1px;
  text-shadow: rgba(0,0,0,.01) 0 0 1px;
}
