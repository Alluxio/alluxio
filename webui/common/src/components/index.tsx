/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export * from './HOCs/withLoadingMessage/withLoadingMessage';
export * from './HOCs/withErrors/withErrors';
export * from './HOCs/withTextAreaResize/withTextAreaResize';
export * from './HOCs/withFetchDataFromPath/withFetchDataFromPath';
export * from './HOCs/withFetchData/withFetchData';
export * from './HOCs/withFluidContainer/withFluidContainer';
export * from './Footer/Footer';
export * from './Header/Header';
export * from './LoadingMessage/LoadingMessage';
export * from './Logs/Logs';
export * from './FileView/FileView';
export * from './Paginator/Paginator';
export * from './LineGraph/LineGraph';
