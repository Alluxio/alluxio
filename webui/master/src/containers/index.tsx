/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

export {default as App} from './App/App';

export {default as Browse} from './pages/Browse/Browse';
export {default as Configuration} from './pages/Configuration/Configuration';
export {default as Data} from './pages/Data/Data';
export {default as Overview} from './pages/Overview/Overview';
export {default as MasterLogs} from './pages/MasterLogs/MasterLogs';
export {default as Workers} from './pages/Workers/Workers';
export {default as Metrics} from './pages/Metrics/Metrics';
