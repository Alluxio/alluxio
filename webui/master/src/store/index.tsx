/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {browseReducer, initialBrowseState} from './browse/reducer';
import {browseSaga} from './browse/sagas';
import {IBrowseState} from './browse/types';
import {configReducer, initialConfigState} from './config/reducer';
import {configSaga} from './config/sagas';
import {IConfigState} from './config/types';
import {dataReducer, initialDataState} from './data/reducer';
import {dataSaga} from './data/sagas';
import {IDataState} from './data/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';
import {IInitState} from './init/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {initialWorkersState, workersReducer} from './workers/reducer';
import {workersSaga} from './workers/sagas';
import {IWorkersState} from './workers/types';

export interface IApplicationState {
  browse: IBrowseState;
  config: IConfigState;
  data: IDataState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
  workers: IWorkersState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  browse: browseReducer,
  config: configReducer,
  data: dataReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history),
  workers: workersReducer
});

export const rootSaga = function* () {
  yield all([
    fork(browseSaga),
    fork(configSaga),
    fork(dataSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga),
    fork(workersSaga)
  ]);
};

export const initialState: IApplicationState = {
  browse: initialBrowseState,
  config: initialConfigState,
  data: initialDataState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState,
  workers: initialWorkersState
};
