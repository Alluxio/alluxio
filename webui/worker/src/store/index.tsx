/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {connectRouter, RouterState} from 'connected-react-router';
import {History} from 'history';
import {combineReducers} from 'redux';
import {all, fork} from 'redux-saga/effects';

import {initialRefreshState, refreshReducer} from '@alluxio/common-ui/src/store/refresh/reducer';
import {IRefreshState} from '@alluxio/common-ui/src/store/refresh/types';
import {initialLogsState, logsReducer} from '@alluxio/common-ui/src/store/logs/reducer';
import {logsSaga} from '@alluxio/common-ui/src/store/logs/sagas';
import {ILogsState} from '@alluxio/common-ui/src/store/logs/types';
import {blockInfoReducer, initialBlockInfoState} from './blockInfo/reducer';
import {blockInfoSaga} from './blockInfo/sagas';
import {IBlockInfoState} from './blockInfo/types';
import {initialMetricsState, metricsReducer} from './metrics/reducer';
import {metricsSaga} from './metrics/sagas';
import {IMetricsState} from './metrics/types';
import {initialOverviewState, overviewReducer} from './overview/reducer';
import {overviewSaga} from './overview/sagas';
import {IOverviewState} from './overview/types';
import {IInitState} from './init/types';
import {initialInitState, initReducer} from './init/reducer';
import {initSaga} from './init/sagas';

export interface IApplicationState {
  blockInfo: IBlockInfoState;
  init: IInitState;
  logs: ILogsState;
  metrics: IMetricsState;
  overview: IOverviewState;
  refresh: IRefreshState;
  router?: RouterState;
}

export const rootReducer = (history: History) => combineReducers<IApplicationState>({
  blockInfo: blockInfoReducer,
  init: initReducer,
  logs: logsReducer,
  metrics: metricsReducer,
  overview: overviewReducer,
  refresh: refreshReducer,
  router: connectRouter(history)
});

export const rootSaga = function* () {
  yield all([
    fork(blockInfoSaga),
    fork(initSaga),
    fork(logsSaga),
    fork(metricsSaga),
    fork(overviewSaga)
  ]);
};

export const initialState: IApplicationState = {
  blockInfo: initialBlockInfoState,
  init: initialInitState,
  logs: initialLogsState,
  metrics: initialMetricsState,
  overview: initialOverviewState,
  refresh: initialRefreshState
};
