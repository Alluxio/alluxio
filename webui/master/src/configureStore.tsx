/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {routerMiddleware} from 'connected-react-router';
import {History} from 'history';
import {applyMiddleware, createStore, Store} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import {IApplicationState, rootReducer, rootSaga} from './store';

export default function configureStore(history: History, initialState: IApplicationState): Store<IApplicationState> {
  const composeEnhancers = composeWithDevTools({});
  const sagaMiddleware = createSagaMiddleware();

  const store = createStore(
    rootReducer(history),
    initialState,
    composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware))
  );

  sagaMiddleware.run(rootSaga);
  return store;
}
