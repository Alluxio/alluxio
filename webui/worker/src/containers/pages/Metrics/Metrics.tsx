/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import React from 'react';
import {connect} from 'react-redux';
import {Progress, Table} from 'reactstrap';
import {compose, Dispatch} from 'redux';

import {withErrors, withFluidContainer, withLoadingMessage, withFetchData} from '@alluxio/common-ui/src/components';
import {IApplicationState} from '../../../store';
import {fetchRequest} from '../../../store/metrics/actions';
import {IMetrics} from '../../../store/metrics/types';
import {createAlertErrors} from "@alluxio/common-ui/src/utilities";
import {ICommonState} from "@alluxio/common-ui/src/constants";

interface IPropsFromState extends ICommonState {
  data: IMetrics
}

interface IPropsFromDispatch {
  fetchRequest: typeof fetchRequest;
}

export type AllProps = IPropsFromState & IPropsFromDispatch;

export class MetricsPresenter extends React.Component<AllProps> {
  public render() {
    const {data} = this.props;

    return (
      <React.Fragment>
        <div className="col-12">
          <h5>Worker Gauges</h5>
          <Table hover={true}>
            <tbody>
            <tr>
              <th scope="row">Worker Capacity</th>
              <td>
                <Progress className="h-50 mt-1" multi={true}>
                  <Progress bar={true} color="dark"
                            value={`${data.workerCapacityFreePercentage}`}>{data.workerCapacityFreePercentage}%
                    Free</Progress>
                  <Progress bar={true} color="secondary"
                            value={`${data.workerCapacityUsedPercentage}`}>{data.workerCapacityUsedPercentage}%
                    Used</Progress>
                </Progress>
              </td>
            </tr>
            </tbody>
          </Table>
        </div>
        <div className="col-12">
          <h5>Logical Operations</h5>
          <Table hover={true}>
            <tbody>
            <tr>
              <th>Blocks Accessed</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksAccessed ? data.operationMetrics.BlocksAccessed.count : 0}</td>
              <th>Blocks Cached</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksCached ? data.operationMetrics.BlocksCached.count : 0}</td>
            </tr>
            <tr>
              <th>Blocks Canceled</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksCanceled ? data.operationMetrics.BlocksCanceled.count : 0}</td>
              <th>Blocks Deleted</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksDeleted ? data.operationMetrics.BlocksDeleted.count : 0}</td>
            </tr>
            <tr>
              <th>Blocks Evicted</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksEvicted ? data.operationMetrics.BlocksEvicted.count : 0}</td>
              <th>Blocks Promoted</th>
              <td>{data.operationMetrics && data.operationMetrics.BlocksPromoted ? data.operationMetrics.BlocksPromoted.count : 0}</td>
            </tr>
            <tr>
              <th>Bytes Read Remotely</th>
              <td>{data.operationMetrics && data.operationMetrics.BytesReadRemote ? data.operationMetrics.BytesReadRemote.count : 0}</td>
              <th>Bytes Written Remotely</th>
              <td>{data.operationMetrics && data.operationMetrics.BytesWrittenRemote ? data.operationMetrics.BytesWrittenRemote.count : 0}</td>
            </tr>
            </tbody>
          </Table>
        </div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = ({metrics, refresh}: IApplicationState): IPropsFromState => ({
  data: metrics.data,
  errors: createAlertErrors(metrics.errors !== undefined),
  loading: metrics.loading,
  refresh: refresh.data,
  class: 'metrics-page'
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  fetchRequest: () => dispatch(fetchRequest())
});

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withFetchData,
  withErrors,
  withLoadingMessage,
  withFluidContainer
)(MetricsPresenter);
