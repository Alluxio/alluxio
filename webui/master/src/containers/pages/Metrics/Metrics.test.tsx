/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createBrowserHistory, History, LocationState} from 'history';
import React from 'react';
import {Provider} from 'react-redux';
import {Store} from 'redux';
import sinon, {SinonSpy} from 'sinon';

import configureStore from '../../../configureStore'
import {initialState, IApplicationState} from '../../../store';
import ConnectedApp from '../../App/App';
import {AllProps, MetricsPresenter} from './Metrics';
import {routePaths} from "../../../constants";
import {createAlertErrors} from "@alluxio/common-ui/src/utilities";

configure({adapter: new Adapter()});

describe('Metrics', () => {
  let history: History<LocationState>;
  let store: Store<IApplicationState>;
  let props: AllProps;

  beforeAll(() => {
    history = createBrowserHistory({keyLength: 0});
    history.push(routePaths.metrics);
    store = configureStore(history, initialState);
    props = {
      data: initialState.metrics.data,
      errors: createAlertErrors(false),
      loading: false,
      refresh: initialState.refresh.data,
      class: '',
      fetchRequest: sinon.spy(() => {})
    };
  });

  afterEach(() => {
    sinon.restore();
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<MetricsPresenter {...props}/>);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });
});
