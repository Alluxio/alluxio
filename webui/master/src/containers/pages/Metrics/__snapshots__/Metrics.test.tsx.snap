// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Metrics Shallow component Matches snapshot 1`] = `
<Fragment>
  <div
    className="col-12"
  >
    <div
      className="row"
    />
  </div>
  <div
    className="col-12"
  >
    <h5>
      Total IO Size
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            Local Alluxio (Domain Socket) Read
          </th>
          <td />
          <th>
            Local Alluxio (Domain Socket) Write
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Local Alluxio (Short-circuit) Read
          </th>
          <td>
            N/A
          </td>
          <th>
            Local Alluxio (Short-circuit) Write
          </th>
          <td>
            N/A
          </td>
        </tr>
        <tr>
          <th>
            Remote Alluxio Read
          </th>
          <td />
          <th>
            Remote Alluxio Write
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Under Filesystem Read
          </th>
          <td />
          <th>
            Under Filesystem Write
          </th>
          <td />
        </tr>
      </tbody>
    </Table>
  </div>
  <div
    className="col-12"
  >
    <h5>
      Total IO Throughput (Last Minute)
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            Local Alluxio (Domain Socket) Read
          </th>
          <td />
          <th>
            Local Alluxio (Domain Socket) Write
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Local Alluxio (Short-circuit) Read
          </th>
          <td>
            N/A
          </td>
          <th>
            Local Alluxio (Short-circuit) Write
          </th>
          <td>
            N/A
          </td>
        </tr>
        <tr>
          <th>
            Remote Alluxio Read
          </th>
          <td />
          <th>
            Remote Alluxio Write
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Under Filesystem Read
          </th>
          <td />
          <th>
            Under Filesystem Write
          </th>
          <td />
        </tr>
      </tbody>
    </Table>
  </div>
  <div
    className="col-12"
  >
    <h5>
      Mounted Under FileSystem Read
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            Under FileSystem
          </th>
          <th>
            Size
          </th>
        </tr>
      </tbody>
    </Table>
  </div>
  <div
    className="col-12"
  >
    <h5>
      Mounted Under FileSystem Write
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            Under FileSystem
          </th>
          <th>
            Size
          </th>
        </tr>
      </tbody>
    </Table>
  </div>
  <div
    className="col-12"
  >
    <h5>
      Logical Operations
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            Directories Created
          </th>
          <td />
          <th>
            File Block Infos Got
          </th>
          <td />
        </tr>
        <tr>
          <th>
            File Infos Got
          </th>
          <td />
          <th>
            Files Completed
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Files Created
          </th>
          <td />
          <th>
            Files Freed
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Files Persisted
          </th>
          <td />
          <th>
            Files Pinned
          </th>
          <td />
        </tr>
        <tr>
          <th>
            New Blocks Got
          </th>
          <td />
          <th>
            Paths Deleted
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Paths Mounted
          </th>
          <td />
          <th>
            Paths Renamed
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Paths Unmounted
          </th>
          <td />
          <th />
          <td />
        </tr>
      </tbody>
    </Table>
  </div>
  <div
    className="col-12"
  >
    <h5>
      RPC Invocations
    </h5>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <tbody>
        <tr>
          <th>
            CompleteFile Operations
          </th>
          <td />
          <th>
            CreateDirectory Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            CreateFile Operations
          </th>
          <td />
          <th>
            DeletePath Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            FreeFile Operations
          </th>
          <td />
          <th>
            GetFileBlockInfo Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            GetFileInfo Operations
          </th>
          <td />
          <th>
            GetNewBlock Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Mount Operations
          </th>
          <td />
          <th>
            RenamePath Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            SetAcl Operations
          </th>
          <td />
          <th>
            SetAttribute Operations
          </th>
          <td />
        </tr>
        <tr>
          <th>
            Unmount Operations
          </th>
          <td />
        </tr>
      </tbody>
    </Table>
  </div>
</Fragment>
`;
